sympy/integrals/integrals.py

```diff
@@ -764,7 +780,7 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
 
         # Piecewise antiderivatives need to call special integrate.
         if f.func is Piecewise:
-            return f._eval_integral(x)
```

Comment:
Is this the source of difference in results? Should the `_eval_integral` be used?